src/devSVG.cpp

```diff
@@ -124,6 +131,37 @@ inline void write_attrs_linetype(FILE* f, int lty, double lwd, int col) {
     fputs("'", f);
     break;
   }
+
+  // Set line end shape
+  switch(lend)
+  {
+  case GE_ROUND_CAP:
+    write_attr_str(f, "stroke-linecap", "round");
+    break;
+  case GE_SQUARE_CAP:
+    write_attr_str(f, "stroke-linecap", "square");
+    break;
+  case GE_BUTT_CAP: // doing nothing, default is butt
+  default:
+    break;
+  }
+
+  // Set line join shape
+  switch(ljoin)
+  {
+  case GE_ROUND_JOIN:
+    write_attr_str(f, "stroke-linejoin", "round");
+    break;
+  case GE_BEVEL_JOIN:
+    write_attr_str(f, "stroke-linejoin", "bevel");
+    break;
+  case GE_MITRE_JOIN: // We don't need to write "stroke-linejoin" attribute here,
+                      // since the default is "miter". However, we do need to
+                      // specify "stroke-miterlimit".
+    write_attr_dbl(f, "stroke-miterlimit", lmitre);
```

Comment:
Hi @hadley ,
I've taken a look at the SVG specification and there is one point that I want to discuss with you.
From your previous suggestion it seems that the proper way to generate line end and line join parameters is the following:

1. First declare a global `<style>` element giving the defaults of `stroke-linecap` and `stroke-linejoin`.
2. When some line styles have non-default values, specify an inline style to override the global one.

However, the precedence of style in SVG is

inline `style` attribute > global `<style>` element > other inline attributes (e.g. `stroke-linecap`)

which means that to override the style declared in `<style>` tag, we have to use `style='stroke-linecap: round;'` instead of `stroke-linecap='round'`. This solution may be a little bit messy, since other properties are specified using their own element attributes, such as `stroke-width` and `fill`, but `stroke-linecap` and `stroke-linejoin` need to be set using the `style` attribute, leading to some SVG code like below:
```
<polyline fill='none' stroke='#000000' style='stroke-linecap: square;' />
```

Do you think this sort of mixed specification of styles is acceptable?